suite/tests/api/ir_riscv64.c

```diff
@@ -1408,6 +1410,260 @@ test_misc(void *dc)
     test_instr_encoding(dc, OP_cbo_inval, instr);
 }
 
+static void
+test_insert_mov_immed_arch(void *dc)
+{
+    instr_t *instr;
+    int i;
+    instrlist_t *ilist = instrlist_create(dc);
+
+    /* Single ADDIW cases. */
+
+#define MOV(imm, encoded)                                                                \
```

Comment:
style: Always wrap a multi-statement macro with `do {...} while (false)` to avoid errors when it's included as a single line in an if or sthg.